File: Java/Lab_23/src/Lab23_2.java
public class Lab23_2 {

    interface IInterface {
        <T1, T2> String getInfo(T1 value1, T2 value2);
    }

    static class CInfo implements IInterface {

        public <T1, T2> String getInfo(T1 value1, T2 value2) {
            return "Value1: " + value1 +
                    ", Type1: " + value1.getClass().getSimpleName() +
                    " | Value2: " + value2 +
                    ", Type2: " + value2.getClass().getSimpleName();
        }
    }

    public static void run() {
        IInterface info = new CInfo();

        System.out.println(info.getInfo(5, "Hello"));
        System.out.println(info.getInfo(3.14, true));
    }
}

File: C#/Lab_23/Lab_23/Lab23_2.cs
namespace Lab_23
{
    public class Lab23_2
    {
        public interface IInterface
        {
            string GetInfo<T1, T2>(T1 value1, T2 value2);
        }

        public class CInfo : IInterface
        {
            public string GetInfo<T1, T2>(T1 value1, T2 value2)
            {
                return $"Value1: {value1}, Type1: {typeof(T1).Name} | " +
                       $"Value2: {value2}, Type2: {typeof(T2).Name}";
            }
        }

        public static void Run()
        {
            IInterface info = new CInfo();
            Console.WriteLine(info.GetInfo(5, "Hello"));
            Console.WriteLine(info.GetInfo(3.14, true));
        }
    }
}
